Develop2

NastyaPilat opened the pull request.

Commit:
new file:   .gitignore
	new file:   conftest.py
	new file:   constants.py
	new file:   tests/__init__.py
	new file:   tests/test_create_order.py
	new file:   tests/test_create_user.py
	new file:   tests/test_get_user_orders.py
	new file:   tests/test_login_user.py
	new file:   tests/test_update_user.py
	new file:   utils.py

Commit:
new file:   requirements.txt

Commit:
new file:   allure_results/23625a95-99b8-457f-9bbf-8fa6bceaaca5-result.json
	new file:   allure_results/3447b4ce-bfab-4834-a6a2-a18c5724e4e1-result.json
	new file:   allure_results/53bcce04-67f6-4479-a89a-ff47f12303c1-result.json
	new file:   allure_results/5aa7d0da-e9f1-46b5-8695-a3adbb6edfa4-result.json
	new file:   allure_results/655ce10d-a5b4-43ee-a0f2-c8d485fe69ba-result.json
	new file:   allure_results/72773020-6656-4980-ae90-83badcb06d41-result.json
	new file:   allure_results/8590f95b-9dc9-486e-8735-cde3d54a1035-result.json
	new file:   allure_results/87233244-104a-4e8b-8647-c72600be0f31-container.json
	new file:   allure_results/a0605752-ba34-4ef5-8f63-1a5a9c895f43-result.json
	new file:   allure_results/a9b2c7c5-c4cd-496b-b6e1-975a03e66e20-result.json
	new file:   allure_results/a9f6526a-03b6-4602-a0a0-fe6955bc4b67-result.json
	new file:   allure_results/ae6d041d-c05a-4a1d-b92e-68cd810495c4-result.json
	new file:   allure_results/b5c28610-354e-47a9-85c2-86785a8739a2-result.json
	new file:   allure_results/c8193d60-8e97-47f1-88de-7df00557da2d-container.json
	new file:   allure_results/d1dfd256-710a-4c2a-b78d-0c931dce8fd9-result.json
	new file:   allure_results/ec4f058c-f0fe-4efa-8ece-e5e9e4e148c9-container.json
	new file:   allure_results/f7358b9e-732d-455c-9ee3-28e4036934b9-result.json
	modified:   tests/test_create_order.py
	modified:   tests/test_create_user.py
	modified:   tests/test_get_user_orders.py
	modified:   tests/test_login_user.py
	modified:   tests/test_update_user.py
	modified:   utils.py

klushcheva commented on constants.py:
```diff
@@ -0,0 +1,6 @@
+BASE_URL = "https://stellarburgers.nomoreparties.site"
+USER_REGISTER_URL = "https://stellarburgers.nomoreparties.site/api/auth/register"
```
Можно лучше: использовать f-string = базовый урл + эндпойнты

klushcheva commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,39 @@
+
+    @allure.title('Создание заказа с авторизацией')
+    def test_create_order_with_auth(self, example_ingredients):
+        user_data = create_new_user()
```
Можно лучше: действия с пользователем вынести в фикстуру, тк нужны почти для каждого теста

klushcheva commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,39 @@
+    @allure.title('Создание заказа с авторизацией')
+    def test_create_order_with_auth(self, example_ingredients):
+        user_data = create_new_user()
+        register_response = requests.post(
```
Можно лучше: создать отдельный класс для формирования запросов, чтобы повысить гибкость и читаемость кода

klushcheva commented on tests/test_create_user.py:
```diff
@@ -0,0 +1,34 @@
+            constants.USER_REGISTER_URL, data=user_data)
+        register_response_2 = requests.post(
+            constants.USER_REGISTER_URL, data=user_data)
+        assert register_response_2.status_code == 403 and not register_response_2.json()['success'] and register_response_2.json()[
```
Можно лучше: стараемся избегать любого хардкода, выносим тексты в отдельный файл

klushcheva reviewed (commented):
Анастасия, хорошая работа, соблюдена архитектура тестов и тестовое покрытие. Есть комментарии, как улучшить код - на твое усмотрение.

Commit:
modified:   conftest.py
	modified:   constants.py
	new file:   request_messages.py
	modified:   tests/test_create_order.py
	modified:   tests/test_create_user.py
	modified:   tests/test_get_user_orders.py
	modified:   tests/test_login_user.py
	modified:   tests/test_update_user.py

Commit:
modified:   conftest.py
	renamed:    constants.py -> endpoints.py
	modified:   tests/test_create_order.py
	modified:   tests/test_create_user.py
	modified:   tests/test_get_user_orders.py
	modified:   tests/test_login_user.py
	modified:   tests/test_update_user.py
